site/src/pages/api/blocks.api.ts

```diff
@@ -105,6 +106,21 @@ export default createApiKeyRequiredHandler<null, ApiSearchResponse>()
         .map(([block, _]) => block);
     }
 
+    for (const block of data) {
+      // Generate absolute URLs for block icons
+      if (block.icon && !block.icon.startsWith("http")) {
+        block.icon = `${FRONTEND_URL}${block.icon}`;
```

Comment:
If the `icon` isn't already an absolute URL, attach the `FRONTEND_URL` to it to generate one.

Reply:
Relative `icon`s are relative to the block's source folder, not the frontend URL - have a look at some examples.